examples/static_graph_optimizations/cifar/train_cifar.py

```diff
@@ -80,19 +94,21 @@ def main():
 
     # Set up a trainer
     updater = training.updaters.StandardUpdater(
-        train_iter, optimizer, device=args.gpu)
+        train_iter, optimizer, device=device)
     trainer = training.Trainer(updater, stop_trigger, out=args.out)
 
     # Evaluate the model with the test dataset for each epoch
-    trainer.extend(extensions.Evaluator(test_iter, model, device=args.gpu))
+    trainer.extend(extensions.Evaluator(test_iter, model, device=device))
 
     # Reduce the learning rate by half every 25 epochs.
     trainer.extend(extensions.ExponentialShift('lr', 0.5),
                    trigger=(25, 'epoch'))
 
     # Dump a computational graph from 'loss' variable at the first iteration
     # The "main" refers to the target link of the "main" optimizer.
-    trainer.extend(extensions.DumpGraph('main/loss'))
+    # TODO(hvy): Support ChainerX
+    if not isinstance(device, chainer.backend.ChainerxDevice):
```

Comment:
`device.xp is chainerx` is better because it's easier to comprehend for users.

Reply:
Fixed.

Reply:
I guess you want to change this style for all xp instances such as https://github.com/chainer/chainer/blob/master/examples/imagenet/train_imagenet.py#L155 ?

Reply:
Perhaps.